ts/sagas/wallet.ts

```diff
@@ -497,46 +503,49 @@ function* completionHandler(_: PaymentRequestCompletion) {
   yield put({ type: PAYMENT_COMPLETED });
 }
 
-function* fetchPagoPaToken(pagoPaClient: PagoPaClient): Iterator<Effect> {
-  const token: Option<string> = yield select(walletTokenSelector);
-  if (token.isSome()) {
-    const response: SagaCallReturnType<
-      typeof pagoPaClient.getSession
-    > = yield call(pagoPaClient.getSession, token.value);
-    if (response !== undefined && response.status === 200) {
-      // token fetched successfully, store it
-      yield put(storePagoPaToken(some(response.value.data.sessionToken)));
-    }
+function* fetchPagoPaToken(
+  pagoPaClient: PagoPaClient,
+  walletToken: string
+): Iterator<Effect> {
+  const response: SagaCallReturnType<
+    typeof pagoPaClient.getSession
+  > = yield call(pagoPaClient.getSession, walletToken);
```

Comment:
for instance, in this case, this saga only needs `pagoPaClient.getSession`, not the whole client